dev-libs/rocksdb/files/rocksdb-6.17.3-libatomic.patch

```diff
@@ -0,0 +1,48 @@
+Fixes check for atomics
+https://bugs.gentoo.org/834855
+
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -316,19 +316,29 @@ endif()
+ # Check if -latomic is required or not
+ if (NOT MSVC)
+   set(CMAKE_REQUIRED_FLAGS "--std=c++11")
+-  CHECK_CXX_SOURCE_COMPILES("
+-#include <atomic>
+-std::atomic<uint64_t> x(0);
+-int main() {
+-  uint64_t i = x.load(std::memory_order_relaxed);
+-  bool b = x.is_lock_free();
+-  return 0;
+-}
+-" BUILTIN_ATOMIC)
+-if (NOT BUILTIN_ATOMIC)
+-  #TODO: Check if -latomic exists
+-  list(APPEND THIRDPARTY_LIBS atomic)
+-endif()
++  set(ATOMIC_TEST_SOURCE "
++	#include <atomic>
++	std::atomic<int> x;
++	std::atomic<short> y;
++	std::atomic<char> z;
++        std::atomic<long long> w;
++	int main() {
++		++z;
++		++y;
++                ++w;
++		return ++x;
++	}")
++  CHECK_CXX_SOURCE_COMPILES("${ATOMIC_TEST_SOURCE}" BUILTIN_ATOMIC)
++  if (NOT BUILTIN_ATOMIC)
++    set(CMAKE_REQUIRED_LIBRARIES atomic)
++    CHECK_CXX_SOURCE_COMPILES("${ATOMIC_TEST_SOURCE}" ATOMICS_REQUIRE_LIBATOMIC)
++    unset(CMAKE_REQUIRED_LIBRARIES)
++    if (ATOMICS_REQUIRE_LIBATOMIC)
++      list(APPEND THIRDPARTY_LIBS atomic)
++    else()
++      message(FATAL_ERROR "Host compiler must support std::atomic!")
++    endif()
++  endif()
+ endif()
+ 
+ # Reset the required flags
```

Comment:
there is already one atomic patch for 6.14.6, so explain a little bit about why we need a new revised version? can all ebuilds using this updated version?